Amplify/Categories/API/Request/RESTOperationRequest.swift

```diff
@@ -30,12 +33,14 @@ public struct RESTOperationRequest: AmplifyOperationRequest {
     public init(apiName: String?,
                 operationType: RESTOperationType,
                 path: String? = nil,
+                headers: [String: String] = [:],
```

Comment:
any reason to do this over `headers: [String: String]? = nil`? it would be more consistent with queryParameters

Reply:
Just to avoid another check for optional. 
I guess, it is better to make it same as query, will update my PR.